src/libotcore/otcore-ed25519-verify.c

```diff
@@ -64,21 +69,24 @@ otcore_validate_ed25519_signature (GBytes *data, GBytes *public_key, GBytes *sig
   // It is OK for error to be NULL, though according to GError rules.
 
 #if defined(HAVE_LIBSODIUM) || defined(HAVE_OPENSSL)
-  // And strictly verify pubkey and signature lengths
-  if (g_bytes_get_size (public_key) != OSTREE_SIGN_ED25519_PUBKEY_SIZE)
-    return glnx_throw (error, "Invalid public key of %" G_GSIZE_FORMAT " expected %" G_GSIZE_FORMAT,
-                       (gsize)g_bytes_get_size (public_key),
-                       (gsize)OSTREE_SIGN_ED25519_PUBKEY_SIZE);
-  const guint8 *public_key_buf = g_bytes_get_data (public_key, NULL);
-  if (g_bytes_get_size (signature) != OSTREE_SIGN_ED25519_SIG_SIZE)
-    return glnx_throw (
-        error, "Invalid signature length of %" G_GSIZE_FORMAT " bytes, expected %" G_GSIZE_FORMAT,
-        (gsize)g_bytes_get_size (signature), (gsize)OSTREE_SIGN_ED25519_SIG_SIZE);
-  const guint8 *signature_buf = g_bytes_get_data (signature, NULL);
+  gsize public_key_size;
+  const guint8 *public_key_buf = g_bytes_get_data (public_key, &public_key_size);
 
+  gsize signature_size;
+  const guint8 *signature_buf = g_bytes_get_data (signature, &signature_size);
 #endif
 
 #if defined(HAVE_LIBSODIUM)
+  // Strictly verify pubkey and signature lengths, as libsodium can
+  // only handle raw ed25519 public key and signatures.
```

Comment:
Right but...it makes me uncomfortable that we now accept arbitrary sized keys and signatures in the openssl case. Does openssl expose any constants that could cap this?

The only things I can really think of are DoS here, and even those feel hard to reach as ostree caps things at `pub const OSTREE_MAX_METADATA_SIZE: c_int = 134217728;`.

Still though...it'd be nice to have *some* sanity check (actually, we could safely cap these things to MAX_METADATA_SIZE) and that way we wouldn't need to special-case libsodium - it would just be the *size* that would vary.


I think in practice, there'

Reply:
Yes, I added the guard unconditionally.